parsers/FR.py

```diff
@@ -139,74 +142,33 @@ def fetch_production(
     # reindex df_production to get 1/2 hourly values
     df_production_reindexed = reindex_data(df_production)
     df_production_reindexed = df_production_reindexed.dropna(how="any", axis=0)
-
-    datapoints = list()
-    for row in df_production_reindexed.iterrows():
-        production = dict()
-        for key, value in MAP_GENERATION.items():
-            if key not in present_fuels:
-                continue
-
-            if -50 < row[1][key] < 0:
-                # set small negative values to 0
-                logger.warning(f"Setting small value of {key} ({value}) to 0.")
-                production[value] = 0
+    df_production_reindexed = df_production_reindexed.rename(columns=MAP_GENERATION)
+    df_production_reindexed = df_production_reindexed.groupby(
+        df_production_reindexed.columns, axis=1
+    ).sum()
+
+    datapoints = ProductionBreakdownList(logger)
```

Comment:
```suggestion
    production_mixes = ProductionBreakdownList(logger)
```
For readability I recommend using clear variables